### DISCUSSION TITLE
How Event Sourcing Architecture and Serverless have super-powered E-commerce systems

### DISCUSSION THREAD

**Discussion by desinole** [dotnetdc/Open-Governance#50] [Topic Ideas / Requests]
@SeanKilleen this is for the March 15th meetup. Assuming evening timeslot

**Title**

How Event Sourcing Architecture and Serverless have super-powered E-commerce systems

**Abstract**

Many business critical systems for e-commerce were built using SOA (Service Oriented Architecture). These days as the volume of data and users have increased exponentially, some of these SOA systems have resorted to desperate measures like stringing together dozens of *synchronous* API calls to fulfill simple operations like inventory look up. This not only results in high cost for performance but also high costs for maintaining these system. This talk attempts to show some of deficiencies of legacy architecture like SOA and how these are addressed in e-commerce by using modern techniques like Event Sourcing. We will heavily leverage the cloud in doing so. As a result, we will use some Serverless principles to handle the plumbing and coding.

Event Sourcing has revolutionized e-commerce by enabling distributed systems on a massive scale. These days some of the best solutions are Serverless - code executing in the cloud, reading and writing data to datastores in the cloud, without having to worry about trivial details like the data schema and scaling issues. In this talk, will walk through an event sourcing architecture and implement an e-commerce use case to demonstrate the power of event sourcing and serverless

**Bio**

Santosh Hari

Sr Azure Cloud & AI Consultant @ Microsoft & Organizer for @Onetug and #OrlandoCC

Santosh is a public speaker who has spoken at tech events including codecamps and regional/international conferences. He is a Sr Azure Cloud & AI Consultant @ Microsoft where he leverages his broad Azure skillset with a DevOps focused approach to add more value to companies looking to migrate to the cloud or build cloud first applications. He was a Microsoft Azure MVP from 2016-2020 and deeply involved with the local community as leader of the Orlando .NET User Group and co-organizer of the Orlando Codecamp. Santosh was listed in TechBeacon's DevOps Top 100 for 2018

**Comment by SeanKilleen** [comment]
Thanks @desinole! I'll update the meetup information shortly.

**Comment by SeanKilleen** [comment]
I've updated issue that pertains to the meetup: https://github.com/dotnetdcmeetup/Open-Governance/issues/1

Meetup link: https://www.meetup.com/dotnetdc/events/284201412/
